opm/simulators/linalg/bda/BdaBridge.cpp

```diff
@@ -191,6 +191,40 @@ void BdaBridge<BridgeMatrix, BridgeVector, block_size>::copySparsityPatternFromI
 }
 
 
+// check if the nnz values of the matrix are in contiguous memory
+// this is done by checking if the distances between addresses of blocks of the first two rows is constant
+template <class BridgeMatrix>
+void checkMemoryContiguous(const BridgeMatrix& mat) {
+    const double *p1 = nullptr, *p2 = nullptr;
+    unsigned distance = 0;
+    for (typename BridgeMatrix::const_iterator r = mat.begin(); r != mat.end(); ++r) {
+        if (r.index() < 2) {
+            for (auto c = r->begin(); c != r->end(); ++c) {
+                auto *p = static_cast<const double*>(&((mat[r.index()][c.index()])[0][0]));
+                if (!p1) {
+                    p1 = p;  // first block
+                } else {
+                    if (!p2) {
+                        p2 = p;  // second block, set distance, future distances should equal this
+                        distance = std::distance(p1, p2);
+                    } else {
+                        // distance already set, now confirm other distance are equal
+                        // update pointers to measure distance next blocks
+                        p1 = p2;
+                        p2 = p;
+                        if (distance != std::distance(p1, p2)) {
+                            OPM_THROW(std::logic_error, "Error memory of Matrix looks not contiguous");
+                        }
+                    }
+                }
+            }
+        } else {
+            break;
+        }
+    }
+}
```

Comment:
I think we can assume that the entries of one row are contiguous. Wouldn't it suffice to only check
```
return mat.rows() < 2 || distance(&(*row[1]->begin())[0][0]), &(*(advance(row[0]->begin(), size_row0-1))[N-1][M-1]) == 1;
```